frontend/src/components/WeatherView.vue

```diff
@@ -0,0 +1,190 @@
+<script setup lang="ts">
+import { ref, onMounted, onUnmounted } from 'vue';
+import axios from 'axios';
+import wsClient from "@/utils/websocket";
+import type { Weather, User } from "@/types/ApiTypes";
+
+const API_DOMAIN = import.meta.env.VITE_API_DOMAIN;
+const users = ref<User[]>([]);
+const retryCounts = ref<{ [key: number]: number }>({});
```

Comment:
I'm not sure you updated this anywhere. I see it referenced in the template, but I don't see any logic to update it.